### PR TITLE
add support for jitpack.io repositories

### PR DESCRIPTION
**Pull request by championswimmer**
Signed-off-by: Arnav Gupta <championswimmer@gmail.com>

### CODE REVIEW

**Comment by championswimmer** [comment]
@jitpack please check and give views as well :)

File: subprojects/core/src/main/groovy/org/gradle/api/internal/artifacts/dsl/DefaultRepositoryHandler.java [hunk 1]
```diff
@@ -37,6 +37,9 @@
     public static final String DEFAULT_BINTRAY_JCENTER_REPO_NAME = "BintrayJCenter";
     public static final String BINTRAY_JCENTER_URL = "https://jcenter.bintray.com/";
 
+    public static final String DEFAULT_JITPACK_REPO_NAME = "JitPack";
+    public static final String JITPACK_URL = "https://jitpack.io"
```
Comment (jitpack-io) [inline subprojects/core/src/main/groovy/org/gradle/api/internal/artifacts/dsl/DefaultRepositoryHandler.java]: What a a great idea! Thank you @championswimmer! Looks like there's a semicolon missing at the end of this line

**Comment by pioterj** [comment]
Thank you for the pull request.

I'm not convinced that this is something that we would like to add to the Gradle API at the moment. We do provide convenience methods for very popular repositories like Maven Central and jCenter. As much as I like the idea behind it, [JitPack](https://jitpack.io/) is nowhere near them in regards to its current [popularity](https://www.google.com/trends/explore#q=jitpack). It's also very easy to use it from Gradle using the current API anyway. 

If you feel strongly about it, feel free to start a discussion on [Gradle Developer List](https://groups.google.com/forum/#!forum/gradle-dev) to get more opinions. It's generally a good idea to do it before submitting any changes. See [Gradle contributor guildelines](https://gradle.org/contribute-to-gradle/) for more information on how to contribute.

**Comment by championswimmer** [comment]
@pioterj Okay, I'll make a post about this on the Google Group. 
Meanwhile let me sign the CLA

**Comment by jitpack-io** [comment]
Won't come as a surprise that we are really excited about this!
```gradle
repositories {
    jitpack()
}
```
Big thanks to Arnav for proposing it! JitPack has always strived to make development easier and this change would definitely add to it. It would mean users can just type the repository and won't have to copy-paste from our website. And it also looks nicer than our current syntax although as Piotr says its quite easy already.

It's understandable that popularity is a concern. JitPack is a fairly young service (less than a year) so that probably explains the search trends page. You can find more usage here on GitHub and it's growing all the time. Incidentally having it in Gradle would add to that popularity:)

Btw, just published a [small script](https://gist.github.com/jitpack-io/380e88da1a5b3efe695b) that enables `jitpack()` repository if anyone wants to give it a quick try.

**Comment by championswimmer** [comment]
Started a discussion on the mailing list here 

https://groups.google.com/forum/#!topic/gradle-dev/AmIYoHn3MQg

**Comment by pioterj** [comment]
As discussed on the mailing list, a better way to add such convenience method is via custom plugin. You can find information on how to do it in my response in that thread. I'm closing this pull request.

**Comment by championswimmer** [comment]
@pioterj 
Any chance of revisiting this ? 
Recently `google()` was added as shorthand function for Google Maven repositories. And since 2015, Jitpack has only risen in popularity.

**Pull request closed without merging**